frontend/src/app/modules/team/teams/user-teams/user-teams.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Team } from '@shared/models/team/team';
+import { TeamService } from '@core/services/team.service';
+import { Observable } from 'rxjs';
+import { ToastNotificationService } from '@core/services/toast-notification.service';
+import { BaseComponent } from '@core/components/base/base.component';
+
+@Component({
+    selector: 'app-user-teams',
+    templateUrl: './user-teams.component.html',
+    styleUrls: ['../teams.component.sass']
+})
+export class UserTeamsComponent extends BaseComponent implements OnInit {
+    @Input() newTeamCreated: Observable<Team> = new Observable<Team>();
+    @Input() joinNewTeam: Observable<Team> = new Observable<Team>();
+
+    @Output() leaveTeamEvent: EventEmitter<Team> = new EventEmitter<Team>();
+
+    @Input() currentOrganizationId: number;
+    @Input() currentUserId: number;
+
+    teams: Team[];
+    isLoading: boolean = false;
+
+    constructor(public teamService: TeamService, private toastService: ToastNotificationService) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.isLoading = true;
+
+        this.teamService
+            .getMemberTeams(this.currentOrganizationId, this.currentUserId)
+            .pipe(this.untilThis)
+            .subscribe(teams => {
+                this.isLoading = false;
+                this.teams = teams.body;
+            }, error => {
+                this.toastService.error(`${error}`, 'Error', 2000);
+            });
+
+        this.joinNewTeam
+            .pipe(this.untilThis)
+            .subscribe(team => {
+                this.teams.push(team);
+                this.toastService.success(`You have successfully joined to #${team.name} team!`, '', 1500);
+            });
+
+        this.newTeamCreated
+            .pipe(this.untilThis)
+            .subscribe(team => {
+                this.teams.push(team);
+            });
+    }
+
+    public leaveTeam(teamId: number) {
+        this.teamService
+            .leaveTeam(teamId, this.currentUserId)
+            .pipe(this.untilThis)
+            .subscribe(response => {
+                this.leaveTeamEvent.emit(response.body);
+                this.teams = this.teams.filter(t => t.id !== teamId);
```

Comment:
I think it would be better to reload teams after deleting

Reply:
okay, but why we can't store cached data ?

Reply:
It is better to have an up-to-date list of teams.
It is a good idea to cache frequent requests but in our case, it isn't.
And also for data that don't change, but in our case not only one user can change this page.
And last but not least if we will have some filters or sorting then it will cause issues because you just push this item but do not apply filters/sorting to it